crates/services/gas_price_service/src/v1/da_source_service/block_committer_costs.rs

```diff
@@ -142,54 +144,309 @@ impl BlockCommitterHttpApi {
         if let Some(url) = &self.url {
-            let val = self.client.get(url).send().await?;
+            let formatted_url = format!("{url}/v1/costs?variant=specific&value={l2_block_number}&limit={PAGE_SIZE}");
+            let val = self.client.get(formatted_url).send().await?;
             tracing::warn!("val: {:?}", val);
-            let response = val.json::<Option<RawDaBlockCosts>>().await?;
+            let response = val.json::<Vec<RawDaBlockCosts>>().await?;
             tracing::warn!("Response: {:?}", response);
             Ok(response)
         } else {
-            Ok(None)
+            Ok(Vec::new())
         }
     }
 
-    async fn get_costs_by_seqno(
-        &self,
-        number: u32,
-    ) -> DaBlockCostsResult<Option<RawDaBlockCosts>> {
+    async fn get_latest_costs(&self) -> DaBlockCostsResult<Option<RawDaBlockCosts>> {
+        // Latest: http://localhost:8080/v1/costs?variant=latest&limit=5
         if let Some(url) = &self.url {
-            let val = self
-                .client
-                .get(format!("{}/{}", url, number))
-                .send()
-                .await?;
+            let formatted_url = format!("{url}/v1/costs?variant=latest&limit=1");
+            let val = self.client.get(formatted_url).send().await?;
             tracing::warn!("val: {:?}", val);
-            let response = val.json::<Option<RawDaBlockCosts>>().await?;
+            let response = val.json::<Vec<RawDaBlockCosts>>().await?;
             tracing::warn!("Response: {:?}", response);
-            Ok(response)
+            Ok(response.first().cloned())
         } else {
             Ok(None)
         }
     }
+}
 
-    async fn get_cost_bundles_by_range(
-        &self,
-        range: core::ops::Range<u32>,
-    ) -> DaBlockCostsResult<Vec<Option<RawDaBlockCosts>>> {
-        if let Some(url) = &self.url {
-            let response = self
-                .client
-                .get(format!("{}/{}-{}", url, range.start, range.end))
-                .send()
-                .await?
-                .json::<Vec<RawDaBlockCosts>>()
-                .await?;
-            Ok(response.into_iter().map(Some).collect())
-        } else {
-            Ok(vec![])
+#[cfg(test)]
+mod test_block_committer_http_api {
+    #![allow(non_snake_case)]
+
+    use super::*;
+    use crate::v1::da_source_service::block_committer_costs::fake_server::FakeServer;
+
+    #[test]
+    fn get_costs_by_l2_block_number__when_url_is_none__then_returns_empty_vec() {
+        let rt = tokio::runtime::Runtime::new().unwrap();
+
+        // given
+        let block_committer = BlockCommitterHttpApi::new(None);
+        let l2_block_number = 1;
+
+        // when
+        let actual = rt.block_on(async {
+            block_committer
+                .get_costs_by_l2_block_number(l2_block_number)
+                .await
+                .unwrap()
+        });
+
+        // then
+        assert_eq!(actual, Vec::new());
```

Comment:
```suggestion
        assert_eq!(actual.len(), 0);
```
will prevent another vec allocation